core/server/api_container/server/startosis_engine/kurtosis_instruction/set_service/set_service.go

```diff
@@ -0,0 +1,247 @@
+	if err != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract value for '%s' argument", ServiceNameArgName)
+	}
+	serviceName := service.ServiceName(serviceNameArgumentValue.GoString())
+
+	builtin.serviceName = serviceName
+
+	serviceConfigOverride, err := builtin_argument.ExtractArgumentValue[*service_config.ServiceConfig](arguments, SetServiceConfigArgName)
+	if err != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract value for '%s' argument", SetServiceConfigArgName)
+	}
+	rawImageVal, found, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[starlark.Value](serviceConfigOverride.KurtosisValueTypeDefault, service_config.ImageAttr)
+	if interpretationErr != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract raw image attribute.")
+	}
+	if !found {
+		return nil, startosis_errors.NewInterpretationError("Unable to extract image attribute off of service config.")
+	}
+	builtin.imageVal = rawImageVal
+	apiServiceConfigOverride, _, interpretationErr := add_service.ValidateAndConvertConfigAndReadyCondition(
+		builtin.serviceNetwork,
+		serviceConfigOverride,
+		locatorOfModuleInWhichThisBuiltInIsBeingCalled,
+		builtin.packageId,
+		builtin.packageContentProvider,
+		builtin.packageReplaceOptions,
+		builtin.imageDownloadMode,
+	)
+	if interpretationErr != nil {
+		return nil, interpretationErr
+	}
+
+	// get existing service config for service and merge with apiServiceConfigOverride
+	currApiServiceConfig, err := builtin.interpretationTimeStore.GetServiceConfig(builtin.serviceName)
+	if err != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "An error occurred retrieving service config for service: %v'", builtin.serviceName)
+	}
+
+	mergedServiceConfig, err := upsertServiceConfigs(currApiServiceConfig, apiServiceConfigOverride)
+	if err != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "An error occurred while overriding service configs in set service for service: %v", builtin.serviceName)
+	}
+
+	builtin.interpretationTimeStore.PutServiceConfig(serviceName, mergedServiceConfig)
+
+	builtin.description = builtin_argument.GetDescriptionOrFallBack(arguments, fmt.Sprintf(descriptionFormatStr, builtin.serviceName))
+	return starlark.None, nil
+}
+
+func (builtin *SetServiceCapabilities) Validate(_ *builtin_argument.ArgumentValuesSet, validatorEnvironment *startosis_validator.ValidatorEnvironment) *startosis_errors.ValidationError {
+	if exists := validatorEnvironment.DoesServiceNameExist(builtin.serviceName); exists == startosis_validator.ComponentNotFound {
+		return startosis_errors.NewValidationError("Service '%v' required by '%v' instruction doesn't exist", builtin.serviceName, SetServiceBuiltinName)
+	}
+	return nil
+}
+
+func (builtin *SetServiceCapabilities) Execute(_ context.Context, _ *builtin_argument.ArgumentValuesSet) (string, error) {
```

Comment:
do i need to do a restart service after this call for this to take effect?

Reply:
No! Similar to `get_service`, the execute is a no op. The effects of this instruction are seen during the execution stage when the `add_service` retrieves the service config from the `interpretationTimeValueStore`.